### PR TITLE
Young

### PR DESCRIPTION
**Pull request by Hogusong**
add message 'saved', 'upgraded', and error message 
set timeout for the message
block others while editing a language.
just add delete button but not delete function yet.

### CODE REVIEW

File: routes/language.js [hunk 1]
```diff
@@ -1,48 +1,64 @@
...
+                Language.update(
+                    {
+                    code: code
+                    },
+                    {
+                    label: label
+                    },
+                    {},
+                    function(err, updated) {
+                    if (err) next(err);
```
Comment (chrislim) [inline routes/language.js]: please fix the spacing here

File: views/songs/language.pug [hunk 2]
```diff
@@ -1,20 +1,38 @@
 extends ../layout
 
 block content
+    -var permit = true
```
Comment (chrislim) [inline views/songs/language.pug]: remove unused variable?

File: views/songs/language.pug [hunk 3]
```diff
@@ -1,20 +1,38 @@
...
+                $(".alert").alert('close');
+            }, 3000);
+        });
+
     ol
         each language in languages
             li(id=`${language.code}`)
               span.display
-                a(href="#" onclick=`editLanguage('${language.code}')`) #{language.label}
+                a(href="#" onclick=`editLanguage('${languages.length}','${language.code}')`) #{language.label}
```
Comment (chrislim) [inline views/songs/language.pug]: why is length being passed in now?

File: views/songs/language.pug [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
+    script(type="text/javascript").
+        $(document).ready(function() {
+            setTimeout(function() {
+                $(".alert").alert('close');
+            }, 3000);
+        });
+
     ol
         each language in languages
             li(id=`${language.code}`)
```
Comment (chrislim) [inline views/songs/language.pug]: add a class='language' attribute

File: views/songs/language.pug [hunk 5]
```diff
@@ -24,12 +42,28 @@ block content
             button.btn.btn-default Add
 
     script(type="text/javascript").
-        function editLanguage (code) {
-          $('#' + code + ' .display').hide();
-          $('#' + code + ' .editor').show();
+        function editLanguage (length, code) {
+          let count = 0;
```
Comment (chrislim) [inline views/songs/language.pug]: A simpler way to get the effect you want:
1. Add a class='language' attribute to the language li
2. Hide all of the language elements $('.language').hide();
3. Show the one that is being edited $('#' + code).show();
4. make it editable 
$('#' + code + ' .display').hide();
$('#' + code + ' .editor').show();

File: views/songs/language.pug [hunk 6]
```diff
@@ -1,21 +1,31 @@
...
+        .alert.alert-danger
+            each error_msg in error_messages
+                p #{error_msg}  
+
     ol
         each language in languages
             li(id=`${language.code}`)
               span.display
-                a(href="#" onclick=`editLanguage('${language.code}')`) #{language.label}
+                a(href="#" onclick=`editLanguage('${languages.length}','${language.code}')`) #{language.label}
```
Comment (chrislim) [inline views/songs/language.pug]: I don't think you need to pass in length anymore.

File: routes/language.js (hunk 1 above)
Comment (chrislim) [reply routes/language.js]: spacing is still off

File: views/songs/language.pug (hunk 3 above)
Comment (Hogusong) [reply views/songs/language.pug]: I forgot to delete it after using 'permit' variable to handle blocking.

File: views/songs/language.pug [hunk 7]
```diff
@@ -24,12 +34,35 @@ block content
...
 
         function cancelEdit (code) {
+          permit = true;
           $('#' + code + ' .display').show();
+          $('.add').show();
           $('.editor').hide();
         }
+
+        function deleteItem (code) {
+          //- here deleting module needed
```
Comment (chrislim) [inline views/songs/language.pug]: is this comment still relevant?

File: views/songs/language.pug [hunk 8]
```diff
@@ -24,12 +34,35 @@ block content
...
+
+          permit = true;
+          $('#' + code + ' .display').show();
+          $('.add').show();
+          $('.editor').hide();          
+        }
+
+        $(document).ready(function() {
+          setTimeout(function() {
+            $(".alert").alert('close');
```
Comment (chrislim) [inline views/songs/language.pug]: use single quote on .alert

**Pull request merged**